core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -734,9 +734,20 @@ public BytesReference ext() {
         return ext;
     }
 
+    /**
+     * Create a new SearchSourceBuilder with attributes set by an xContent.
+     */
     public SearchSourceBuilder fromXContent(XContentParser parser, QueryParseContext context, AggregatorParsers aggParsers)
             throws IOException {
         SearchSourceBuilder builder = new SearchSourceBuilder();
+        builder.parseXContent(parser, context, aggParsers);
```

Comment:
More fiddly request handling stuff. I don't feel to bad about this because the ObjectParser interface uses this "parse into object style" which I'm adding here for reindex.